Wrappers/Python/cil/optimisation/functions/TotalGeneralisedVariation.py

```diff
@@ -0,0 +1,211 @@
+        the composition of a separable function 
+        
+        .. math:: f(z_{1}, z_{2}) = f_{1}(z_{1}) + f_{2}(z_{2}) = \alpha\|z_{1}\|_{2,1} + \beta\|z_{2}\|_{2,1}
+
+        and the operator
+
+        .. math:: K = \begin{bmatrix}
+                    \nabla & -\mathbb{I}\\
+                    \mathbb{O} & \mathcal{E}
+                  \end{bmatrix}
+        
+        Therefore, 
+
+        .. math:: f(K \begin{bmatrix}
+                       u \\
+                       w 
+                     \end{bmatrix}) = f_{1}(\nabla u - w) + f_{2}(\mathcal{E}w)
+        
+
+        In that case, the proximal operator of TGV does not have an exact solution and we use an iterative 
+        algorithm to solve:
+
+
+        .. math:: \mathrm{prox}_{\tau \mathrm{TGV}_{\alpha,\beta}}(b) := \underset{u}{\mathrm{argmin}} \frac{1}{2\tau}\|u - b\|^{2} + \mathrm{TGV_{\alpha, \beta}}(u) \Leftrightarrow
+
+        .. math:: \underset{u,w}{\mathrm{argmin}} \frac{1}{2\tau}\|u - b\|^{2} +  \alpha \|\nabla u - w\|_{2,1} + \beta\|\mathcal{E}w\|_{2,1} 
+
+        
+        The algorithm used for the proximal operator of TGV is the Primal Dual Hybrid Algorithm, see :class:`.PDHG`.
+
+
+        Parameters
+        ----------
+        max_iteration : :obj:`int`, default = 100
+            Maximum number of iterations for the PDHG algorithm.
+        correlation : :obj:`str`, default = `Space`
+            Correlation between `Space` and/or `SpaceChannels` for the :class:`.GradientOperator`.
+        backend :  :obj:`str`, default = `c`      
+            Backend to compute the :class:`.GradientOperator`  
+        split : :obj:`boolean`, default = False
+            Splits the Gradient into spatial gradient and spectral or temporal gradient for multichannel data.
+        verbose : :obj:`int`, default = 0
+            Prints primal, dual and primal-dual objectives of the :class:`PDHG` algorithm.
+        update_objective_interval : :obj:`int`, default = :code:`max_iteration`
+            Updates the objectives of the :class:`PDHG` algorithm.
+
+
+        Examples
+        --------
+
+        .. math:: \mathrm{prox}_{\tau\mathrm{TGV}_{\alpha, \beta}}(b) = \underset{u}{\mathrm{argmin}} \frac{1}{2\tau}\|u - b\|^{2} + \mathrm{TGV}_{\alpha, \beta}(u) 
+
+
+        >>> alpha1 = 1.0
+        >>> alpha0 = 2.0
+        >>> TGV = TotalGeneralisedVariation(alpha1 = alpha1, alpha0 = alpha0)
+        >>> sol = TGV.proximal(b, tau = 1.0)
+
+
+        """   
+        
+            
+    def __init__(self,
+                 reg_parameter=1.0,
+                 alpha1 = 1.0,
+                 alpha0 = 2.0,
+                 max_iteration = 100, 
+                 update_objective_interval = 50,
+                 correlation = "Space",
+                 backend = "c",
+                 split = False,
+                 verbose = 0,                  
+                 warmstart=False):
```

Comment:
PEP8 says
> Don’t use spaces around the = sign when used to indicate a keyword argument, or when used to indicate a default value for an unannotated function parameter:

I'd use autoyapf with 120 character line to get this fixed. It is an extension to VSCode